fetch.bs

```diff
@@ -4387,12 +4387,18 @@ steps:
 
       <p class=note>This covers the case of <var>response</var> being a <a>network error</a>.
 
+     <li><p>Let <var>responseStatus</var> be 0 if <var>fetchParams</var>'s
+     <a for="fetch params">request</a>'s <a for=request>mode</a> is "<code>navigate</code>" and
+     <var>response</var>'s <a for=response>URL</a>'s <a for=url>origin</a> is not
+     <a>same origin</a> with <var>request</var>'s <a for=request>origin</a>, and
+     <var>response</var>'s <a for=response>status</a> otherwise.
```

Comment:
I don't think the same origin check here is good. This doesn't take redirects into account.

Also, is this the internal response or not?

Reply:
To take redirects into acount, would introducing a new flag to request be a good idea - `is navigation same origin`. As [HTTP fetch step 6.6](https://fetch.spec.whatwg.org/#ref-for-timing-allow-failed%E2%91%A2) we would check if request’s [mode](https://fetch.spec.whatwg.org/#concept-request-mode) is "navigate" and request’s [current URL](https://fetch.spec.whatwg.org/#concept-request-current-url)’s [origin](https://url.spec.whatwg.org/#concept-url-origin) is [same origin](https://html.spec.whatwg.org/multipage/origin.html#same-origin) with request’s [origin](https://fetch.spec.whatwg.org/#concept-request-origin). If at any time this is not the case we set `is navigation same origin` to false.

This flag would then be used instead of the current check. Would this work?

Reply:
Can we use https://fetch.spec.whatwg.org/#concept-request-tainted-origin? I think request's URL list is still being build up by HTML's navigate algorithm so that should be fine. cc @noamr

Reply:
Okay, so just to be clear we let `responseStatus` be 0 if request’s [mode](https://fetch.spec.whatwg.org/#concept-request-mode) is "navigate" and request has a [redirect-tainted origin](https://fetch.spec.whatwg.org/#concept-request-tainted-origin)?

Also, I believe Noam is OOO for a few months.


> Also, is this the internal response or not?

I don't think it is the internal response. We want the filtered response's status, so that status would be 0 if its say a no-cors request as well.

Reply:
I think so, but now I wonder why this is different from the other measurements. If this is specifically for navigation timing it should probably follow the current model for those. cc @noamr

Reply:
This is not about navigation timing but about iframe navigations being reported to their parent. Since they cannot be CORS fetches, we don't want to report their status if they are cross origin.

Reply:
I don't believe we do, but I could be wrong. Are you aware of such infrastructure?

Reply:
Wait, we have https://html.spec.whatwg.org/multipage/browsing-the-web.html#navigation-params-has-cross-origin-redirects for NavTiming. I can see if/how it fits here.